packages/compass-crud/src/components/bulk-update-dialog.tsx

```diff
@@ -0,0 +1,298 @@
+  },
+  '& .cm-editor': {
+    background: 'transparent !important',
+  },
+});
+
+const bannerContainerStyles = css({
+  // don't jump when an error appears
+  minHeight: spacing[4] * 2 + 2,
+  overflow: 'hidden',
+});
+
+const bannerStyles = css({
+  flex: 'none',
+  marginTop: spacing[2],
+  marginLeft: spacing[2],
+  marginRight: spacing[2],
+  textAlign: 'left',
+});
+
+const updatePreviewStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[3],
+});
+
+export type BulkUpdateDialogProps = {
+  isOpen: boolean;
+  ns: string;
+  filter: BSONObject;
+  count?: number;
+  updateText: string;
+  preview: UpdatePreview;
+  syntaxError?: Error & { loc?: { index: number } };
+  serverError?: Error;
+  closeBulkUpdateDialog: () => void;
+  updateBulkUpdatePreview: (updateText: string) => void;
+  runBulkUpdate: () => void;
+};
+
+export default function BulkUpdateDialog({
+  isOpen,
+  ns,
+  filter,
+  count,
+  updateText,
+  preview,
+  syntaxError,
+  serverError,
+  closeBulkUpdateDialog,
+  updateBulkUpdatePreview,
+  runBulkUpdate,
+}: BulkUpdateDialogProps) {
+  const darkMode = useDarkMode();
+
+  const [text, setText] = useState(updateText);
+
+  const previewDocuments = useMemo(() => {
+    return preview.changes.map(
+      (change) => new HadronDocument(change.after as Record<string, unknown>)
+    );
+  }, [preview]);
+
+  const onChangeText = (value: string) => {
+    setText(value);
+    updateBulkUpdatePreview(value);
+  };
+
+  const title =
+    count === undefined ? 'Update documents' : `Update documents (${count})`;
+
+  const annotations = useMemo<Annotation[]>(() => {
+    if (syntaxError?.loc?.index) {
+      return [
+        {
+          message: syntaxError.message,
+          severity: 'error',
+          from: syntaxError.loc.index,
+          to: syntaxError.loc.index,
+        },
+      ];
+    }
+
+    return [];
+  }, [syntaxError]);
+
+  return (
+    <FormModal
+      title={title}
+      subtitle={ns}
+      size="large"
+      open={isOpen}
+      onSubmit={runBulkUpdate}
+      onCancel={closeBulkUpdateDialog}
+      cancelButtonText="Close"
+      submitButtonText="Update documents"
+      submitDisabled={!!(syntaxError || serverError)}
+    >
+      <div className={columnsStyles}>
+        <div className={queryStyles}>
+          <div className={queryFieldStyles}>
+            <div className={filterLabelContainerStyles}>
+              <Label htmlFor="bulk-update-filter">Filter</Label>
+              <InfoSprinkle>
+                Return to the Documents tab to edit this query
+              </InfoSprinkle>
+            </div>
+            <KeylineCard
+              className={cx(
+                codeContainerStyles,
+                inlineContainerStyles,
+                darkMode ? codeDarkContainerStyles : codeLightContainerStyles
+              )}
+            >
+              <pre>{(toJSString(filter) ?? '').replace(/\s+/g, ' ')}</pre>
```

Comment:
Need to test and do something about wide filters that don't fit. I think the plan is to cut them off and show the full thing in a tooltip. As opposed to wrapping or scrolling.

Reply:
I just copied the same pattern/style from the bulk delete modal: just use a text input.